packages/components/src/components/single-select/shadow.tsx

```diff
@@ -0,0 +1,704 @@
+
+	/**
+	 * Hides the error message but leaves it in the DOM for the input's aria-describedby.
+	 * @TODO: Change type back to `HideErrorPropType` after Stencil#4663 has been resolved.
+	 */
+	@Prop({ mutable: true, reflect: true }) public _hideError?: boolean = false;
+
+	/**
+	 * Hides the caption by default and displays the caption text with a tooltip when the
+	 * interactive element is focused or the mouse is over it.
+	 * @TODO: Change type back to `HideLabelPropType` after Stencil#4663 has been resolved.
+	 */
+	@Prop() public _hideLabel?: boolean = false;
+
+	/**
+	 * Defines the hint text.
+	 */
+	@Prop() public _hint?: string = '';
+
+	/**
+	 * Defines the icon classnames (e.g. `_icons="fa-solid fa-user"`).
+	 */
+	@Prop() public _icons?: Stringified<KoliBriHorizontalIcons>;
+
+	/**
+	 * Defines the internal ID of the primary component element.
+	 */
+	@Prop() public _id?: IdPropType;
+
+	/**
+	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.). Set to `false` to enable the expert slot.
+	 */
+	@Prop() public _label!: LabelWithExpertSlotPropType;
+
+	/**
+	 * Defines the properties for a message rendered as Alert component.
+	 */
+	@Prop() public _msg?: MsgPropType;
+
+	/**
+	 * Defines the technical name of an input field.
+	 */
+	@Prop() public _name?: NamePropType;
+
+	/**
+	 * Gibt die EventCallback-Funktionen für das Input-Event an.
+	 */
+	@Prop() public _on?: InputTypeOnDefault;
+
+	/**
+	 * Options the user can choose from.
+	 */
+	@Prop() public _options!: OptionsWithOptgroupPropType;
+
+	/**
+	 * Makes the input element required.
+	 * @TODO: Change type back to `RequiredPropType` after Stencil#4663 has been resolved.
+	 */
+	@Prop() public _required?: boolean = false;
+
+	/**
+	 * Selector for synchronizing the value with another input element.
+	 * @internal
+	 */
+	@Prop() public _syncValueBySelector?: SyncValueBySelectorPropType;
+
+	/**
+	 * Defines which tab-index the primary element of the component has. (https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex)
+	 */
+	@Prop() public _tabIndex?: number;
+
+	/**
+	 * Defines where to show the Tooltip preferably: top, right, bottom or left.
+	 */
+	@Prop() public _tooltipAlign?: TooltipAlignPropType = 'top';
+
+	/**
+	 * Shows if the input was touched by a user.
+	 * @TODO: Change type back to `TouchedPropType` after Stencil#4663 has been resolved.
+	 */
+	@Prop({ mutable: true, reflect: true }) public _touched?: boolean = false;
+
+	/**
+	 * Defines the value of the input.
+	 */
+	@Prop({ mutable: true }) public _value?: string;
+
+	@State() public state: SingleSelectStates = {
+		_hasValue: false,
+		_hideError: false,
+		_id: `id-${nonce()}`,
+		_label: '', // ⚠ required
+		_options: [],
+		_value: '',
+	};
+
+	public constructor() {
+		this.controller = new SingleSelectController(this, 'select', this.host);
+		this.onInput = this.onInput.bind(this);
```

Comment:
Ich finde die Lösung gut, allerdings wird `this.onInput` an mindestens einer Stelle doppelt "binded". Aus Gründen der Konsistenz würde ich `onChange`, `toggleListbox` und `handleKeyDownDropdown` genauso behandeln.

Reply:
Es gibt immer noch einige doppelte `bind`s, such mal nach `.bind(` in der Datei.